hooks.md

```diff
@@ -0,0 +1,92 @@
+
+
+
+### Json Definition
+
+We decided to add a json file for hook builders which allows them to tell CRI-O when to run the hook and in which stage.
+CRI-O reads all of the json files in /usr/share/containers/oci/hooks.d/*.json and /etc/containers/oci/hooks.d and
+then set up these hooks to run.  If the same name is in both directories, the one in /etc/containers/oci/hooks.d will be
+used.
+
+The json configuration looks like this in GO
+```
+// HookParams is the structure returned from read the hooks configuration
+type HookParams struct {
+	Hook          string   `json:"hook"`
+	Stage         []string `json:"stages"`
+	Cmds          []string `json:"cmds"`
+	Annotations   []string `json:"annotations"`
+	HasBindMounts bool     `json:"hasbindmounts"`
+}
+```
+
+| Key    | Description                                                                                                                        | Required/Optional |
+| ------ |----------------------------------------------------------------------------------------------------------------------------------- | -------- |
+| hook   | Path to the hook                                                                                                                   | Required |
+| stages | List of stages to run the hook in: Valid options are `prestart`, `poststart`, `poststop`                                           | Required |
+| cmds   | List of regular expressions to match the command for running the container.  If the command matches a regex, the hook will be run  | Optional |
+| annotations | List of regular experssions to match against the Annotations in the container runtime spec, if an Annotations matches the hook will be run|optional |
+| hasbindmounts | Tells CRI-O to run the hook if the container has bind mounts from the host into the container | Optional |
+
+### Example
+
+
+```
+cat /etc/containers/oci/hooks.d/oci-systemd-hook.json
+{
+    "cmds": [".*/initi$" , ".*/systemd$" ],
+    "hook": "/usr/libexec/oci/hooks.d/oci-systemd-hook",
+    "stages": [ "prestart", "poststop" ]
+}
+```
+
+In the above example CRI-O will only run the oci-systemd-hook in the prestart and poststop stage, iff the command ends with /init or /systemd
```

Comment:
suggest "iff" to "if and only if" or just "if"

Reply:
fixed with if.